src/Cmfd.h

```diff
@@ -70,29 +63,28 @@ class Cmfd {
   FP_PRECISION _k_eff;
 
   /** The A (destruction) matrix */
-  FP_PRECISION** _A;
+  Matrix* _A;
 
   /** The M (production) matrix */
-  FP_PRECISION** _M;
+  Matrix* _M;
 
   /** The old source vector */
-  FP_PRECISION* _old_source;
+  Vector* _old_source;
 
   /** The new source vector */
-  FP_PRECISION* _new_source;
+  Vector* _new_source;
 
   /** Vector representing the flux for each cmfd cell and cmfd enegy group at 
    * the end of a CMFD solve */
-  FP_PRECISION* _new_flux;
+  Vector* _new_flux;
 
   /** Vector representing the flux for each cmfd cell and cmfd enegy group at 
    * the beginning of a CMFD solve */
-  FP_PRECISION* _old_flux;
-
-  /** Vector representing the flux during the previous iteration of a 
-   * cmfd solve */
-  FP_PRECISION* _flux_temp;
+  Vector* _old_flux;
```

Comment:
This vector class is convenient because it allows us to set values given a cell and energy group, which makes things a bit cleaner. Previously, one would have to compute the index by multiplying the cell by the number of groups each time we want to set/increment a value. This vector class also has thread-safe set/increments, so it is kind of like your replacement of the unordered_maps with the parallel hash maps.